### PR TITLE
Fix: pre-render the nav sidebar

### PR DESCRIPTION
**Pull request by katspaugh**
## What it solves

Partly #1714

## How this PR fixes it

The sidebar was previously rendered conditionally in two variants: for `?safe` routes and for non-safe routes.
Since we now don't show the sidebar for pages like Welcome, Safe Creation etc, there's no need for this variance, and we can always render the nav sidebar for a better static site performance.

## How to test it
Disable JS and load the Balances page of any Safe. The sidebar should appear much faster.

### CODE REVIEW

File: src/components/sidebar/Sidebar/index.tsx [hunk 1]
```diff
@@ -11,54 +10,49 @@ import SidebarFooter from '@/components/sidebar/SidebarFooter'
...
-  const onDrawerToggle = () => {
-    trackEvent({ ...OVERVIEW_EVENTS.SIDEBAR, label: isDrawerOpen ? 'Close' : 'Open' })
-    setIsDrawerOpen((prev) => !prev)
-  }
+  const onDrawerToggle = useCallback(() => {
+    setIsDrawerOpen((isOpen) => {
+      trackEvent({ ...OVERVIEW_EVENTS.SIDEBAR, label: isOpen ? 'Close' : 'Open' })
+
+      return !isOpen
+    })
```
Comment (iamacook) [inline src/components/sidebar/Sidebar/index.tsx:22]: Why was this change necessary?

Comment (iamacook) [inline src/components/sidebar/Sidebar/index.tsx:22]: Why was this change necessary?

**Review by iamacook — APPROVED** [review]

File: src/components/sidebar/Sidebar/index.tsx (hunk 1 above)
Comment (katspaugh) [reply src/components/sidebar/Sidebar/index.tsx:22]: To not create a new callback reference on each render.

**Review by usame-algan — APPROVED** [review]

**Comment by katspaugh** [comment]
I've tested it and since it doesn't touch any logic, going to merge it directly.
